src/components/select/select-list/select-list.component.js

```diff
@@ -196,20 +199,38 @@ const SelectList = React.forwardRef(
       updateListScrollTop(indexOfMatch, listRef.current);
     }, [getIndexOfMatch, highlightedValue]);
 
-    function getChildrenWithListProps() {
-      return React.Children.map(children, (child, index) => {
+    const handleSelect = useCallback(
+      (optionData) => {
+        onSelect({ ...optionData, selectionType: "click" });
+      },
+      [onSelect]
+    );
+
+    const getChildrenWithListProps = useCallback(() => {
+      const optionsList = React.Children.map(children, (child, index) => {
+        if (child.type !== Option) {
+          return child;
+        }
+
         const newProps = {
           onSelect: handleSelect,
           isHighlighted: currentOptionsListIndex === index,
+          hidden: isLoading,
         };
 
         return React.cloneElement(child, newProps);
       });
-    }
 
-    function handleSelect(optionData) {
-      onSelect({ ...optionData, selectionType: "click" });
-    }
+      if (isLoading) {
+        optionsList.push(
```

Comment:
If we return here instead of pushing to the `optionList` we don't need to support `hidden` on the option component.

If we do that we can:
- skip the cloning of all the children elements that are set to `display:none`.
- remove the code that allows an option to be hidden.

Reply:
by doing that I wanted to avoid deselection of previously set option, you're right, it will not affect the children prop, so it won't trigger a re-render of the parent.

Reply:
the loader will have to be pushed as the last element when the lazy loading is implemented. I could remove the `hidden` prop and clear the list when that prop is set to true.